conans/client/new.py

```diff
@@ -111,20 +111,20 @@ class {package_name}TestConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
     requires = "{name}/{version}@%s/%s" % (username, channel)
     generators = "cmake"
+    cmake = None
 
     def build(self):
-        cmake = CMake(self.settings)
+        self.cmake = CMake(self.settings)
         # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is in "test_package"
-        cmake.configure(self, source_dir=self.conanfile_directory, build_dir="./")
-        cmake.build(self)
+        self.cmake.configure(self, source_dir=self.conanfile_directory, build_dir="./")
+        self.cmake.build(self)
 
     def imports(self):
         self.copy("*.dll", "bin", "bin")
         self.copy("*.dylib", "bin", "bin")
 
     def test(self):
-        os.chdir("bin")
-        self.run(".%sexample" % os.sep)
+        self.cmake.test(self)
```

Comment:
Yes, please revert it.
It is more a communication issue, we want to make sure that conan newbies understand the purpose, and what the ``test()`` method is doing, and both coupling it with the ``build()`` method (as the cmake object is created there) and hiding the execution of the binary, makes it less explicit.

So maybe the best is to revert it to the previous state and leave the cmake.test call in comments, but with a new instance:
```cmake
# cmake = CMake(self.settings)
# cmake.test(self)
```

Also leaving in comments the ``enable_testing()`` and test related cmake lines in ``CMakeLists.txt``. In this way it will be both explicit and with good hints for those willing to use the CMake features.